code/game/machinery/vending/cm_vending.dm

```diff
@@ -1165,58 +1163,58 @@ GLOBAL_LIST_INIT(cm_vending_gear_corresponding_types_list, list(
 	.["stock_listing"] = stock_values
 	.["current_m_points"] = points
 
-/proc/vendor_successful_vend(obj/structure/machinery/cm_vending/vendor, list/itemspec, mob/living/carbon/human/user)
-	if(vendor.stat & IN_USE)
+/obj/structure/machinery/cm_vending/proc/vendor_successful_vend(list/itemspec, mob/living/carbon/human/user)
+	if(stat & IN_USE)
 		return
-	vendor.stat |= IN_USE
+	stat |= IN_USE
 
-	var/vend_flags = vendor.vend_flags
-	var/turf/target_turf = vendor.get_appropriate_vend_turf(user)
+	var/vendor_flags = vend_flags
+	var/turf/target_turf = get_appropriate_vend_turf(user)
 	if(LAZYLEN(itemspec)) //making sure it's not empty
-		if(vendor.vend_delay)
-			vendor.overlays.Cut()
-			vendor.icon_state = "[initial(vendor.icon_state)]_vend"
-			if(vendor.vend_sound)
-				playsound(vendor.loc, vendor.vend_sound, 25, 1, 2) //heard only near vendor
-			sleep(vendor.vend_delay)
+		if(vend_delay)
+			overlays.Cut()
+			icon_state = "[initial(icon_state)]_vend"
+			if(vend_sound)
+				playsound(loc, vend_sound, 25, 1, 2) //heard only near vendor
+			sleep(vend_delay)
 
 		var/prod_type = itemspec[3]
 		if(islist(prod_type))
 			for(var/each_type in prod_type)
-				vendor_successful_vend_one(vendor, each_type, user, target_turf, itemspec[4] == MARINE_CAN_BUY_UNIFORM)
+				vendor_successful_vend_one(each_type, user, target_turf, itemspec[4] == MARINE_CAN_BUY_UNIFORM)
 		else
-			vendor_successful_vend_one(vendor, prod_type, user, target_turf, itemspec[4] == MARINE_CAN_BUY_UNIFORM)
+			vendor_successful_vend_one(prod_type, user, target_turf, itemspec[4] == MARINE_CAN_BUY_UNIFORM)
 
-		if(vend_flags & VEND_LIMITED_INVENTORY)
+		if(vendor_flags & VEND_LIMITED_INVENTORY)
 			itemspec[2]--
-			if(vend_flags & VEND_LOAD_AMMO_BOXES)
-				vendor.update_derived_ammo_and_boxes(itemspec)
+			if(vendor_flags & VEND_LOAD_AMMO_BOXES)
+				update_derived_ammo_and_boxes(itemspec)
 
 	else
 		to_chat(user, SPAN_WARNING("ERROR: itemspec is missing. Please report this to admins."))
 		sleep(15)
 
-	vendor.stat &= ~IN_USE
-	vendor.icon_state = initial(vendor.icon_state)
-	vendor.update_icon()
+	stat &= ~IN_USE
+	icon_state = initial(icon_state)
+	update_icon()
 
-/proc/vendor_successful_vend_one(obj/structure/machinery/cm_vending/vendor, prod_type, mob/living/carbon/human/user, turf/target_turf, insignas_override)
+/obj/structure/machinery/cm_vending/proc/vendor_successful_vend_one(prod_type, mob/living/carbon/human/user, turf/target_turf, insignas_override)
 	var/obj/item/new_item
-	var/vend_flags = vendor.vend_flags
+	var/vendor_flags = vend_flags
```

Comment:
we can probably just use vend_flags given this var is never modified

Reply:
Done.